src/connector/src/sink/mongodb.rs

```diff
@@ -0,0 +1,860 @@
+                )));
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn new_log_sinker(&self, writer_param: SinkWriterParam) -> Result<Self::LogSinker> {
+        Ok(MongodbSinkWriter::new(
+            format!("{}-{}", writer_param.executor_id, self.param.sink_name),
+            self.config.clone(),
+            self.schema.clone(),
+            self.pk_indices.clone(),
+            self.is_append_only,
+        )
+        .await?
+        .into_log_sinker(writer_param.sink_metrics))
+    }
+}
+
+pub struct MongodbSinkWriter {
+    pub config: MongodbConfig,
+    client: ClientGuard,
+    payload_writer: MongodbPayloadWriter,
+}
+
+impl MongodbSinkWriter {
+    pub async fn new(
+        name: String,
+        config: MongodbConfig,
+        schema: Schema,
+        pk_indices: Vec<usize>,
+        is_append_only: bool,
+    ) -> Result<Self> {
+        let client = config.common.build_client().await?;
+
+        let default_namespace =
+            config
+                .common
+                .collection_name
+                .parse()
+                .map_err(|err: mongodb::error::Error| {
+                    SinkError::Mongodb(anyhow!(err).context("parsing default namespace failed"))
+                })?;
+
+        let coll_name_field_index =
+            config
+                .collection_name_field
+                .as_ref()
+                .and_then(|coll_name_field| {
+                    schema
+                        .names_str()
+                        .iter()
+                        .position(|&name| coll_name_field == name)
+                });
+
+        let payload_writer = MongodbPayloadWriter::new(
+            schema,
+            pk_indices,
+            default_namespace,
+            coll_name_field_index,
+            config.drop_collection_name_field,
+            is_append_only,
+            client.clone(),
+            config.bulk_write_max_entries,
+        );
+
+        Ok(Self {
+            config,
+            client: ClientGuard::new(name, client),
+            payload_writer,
+        })
+    }
+}
+
+#[async_trait]
+impl SinkWriter for MongodbSinkWriter {
+    async fn begin_epoch(&mut self, _epoch: u64) -> Result<()> {
+        Ok(())
+    }
+
+    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
+        for (op, row) in chunk.rows() {
+            self.payload_writer.write(op, row).await?;
+        }
+        Ok(())
+    }
+
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<Self::CommitMetadata> {
+        if is_checkpoint {
+            self.payload_writer.flush().await?;
+        }
+        Ok(())
+    }
+}
+
+/// We support converting all types to `MongoDB`. If there is an unmatched type, it will be
+/// converted to its string representation. If there is an conversion error, a warning log is printed
+/// and a `Bson::Null` is returned
+fn bson_from_scalar_ref<'a>(field: &'a Field, datum: Option<ScalarRefImpl<'a>>) -> Bson {
```

Comment:
Do you mind moving this method to a separate file under `src/connector/src/sink/encoder` as an implementation of `BsonEncoder`, similar to the `impl RowEncoder for JsonEncoder`?

Reply:
It's feasible, let me try it. This will make `mongodb.rs` a little bit shorter.